src/XMakeBuildEngine/UnitTestsPublicOM/Definition/Project_Tests.cs

```diff
@@ -2415,6 +2416,251 @@ public void ImportPropertyEvaluatingToInvalidPath()
+        {
+            var project =
+                @"<Project ToolsVersion='msbuilddefaulttoolsversion' DefaultTargets='Build' xmlns='msbuildnamespace'>
+                  <ItemGroup>
+                    <B Include=`1;2;3`/>
+                    <A Include=`*;$(P);@(B)`/>
+                  </ItemGroup>
+
+                  <PropertyGroup>
+                    <P>@(B)</P>  
+                  </PropertyGroup>
+                </Project>
+                ";
+
+            var expected = new List<Tuple<string, Operation, Provenance, int>>
+            {
+                Tuple.Create("B", Operation.Include, Provenance.StringLiteral, 1),
+                Tuple.Create("A", Operation.Include, Provenance.Inconclusive | Provenance.Glob, 3)
+            };
+
+            AssertProvenanceResult(expected, project, "1");
+        }
+
+        [Fact]
+        public void GetItemProvenanceMatchesLiteralsWithNonCanonicPaths()
+        {
+            var project =
+                @"<Project ToolsVersion='msbuilddefaulttoolsversion' DefaultTargets='Build' xmlns='msbuildnamespace'>
+                  <ItemGroup>
+                    <A Include=`1.foo;.\1.foo;.\.\1.foo`/>
+                  </ItemGroup>
+                </Project>
+                ";
+
+            var expected = new List<Tuple<string, Operation, Provenance, int>>
+            {
+                Tuple.Create("A", Operation.Include, Provenance.StringLiteral, 3)
+            };
+
+            AssertProvenanceResult(expected, project, "1.foo");
+            AssertProvenanceResult(expected, project, @".\1.foo");
+        }
+
+        private static void AssertProvenanceResult(List<Tuple<string, Operation, Provenance, int>> expected, string project, string itemValue)
+        {
+            var provenanceResult = ObjectModelHelpers.CreateInMemoryProject(project).GetItemProvenance(itemValue);
+            AssertProvenanceResult(expected, provenanceResult);
+        }
+
+        private static void AssertProvenanceResult(List<Tuple<string, Operation, Provenance, int>> expected, string project, string itemValue, string itemType)
+        {
+            var provenanceResult = ObjectModelHelpers.CreateInMemoryProject(project).GetItemProvenance(itemValue, itemType);
+            AssertProvenanceResult(expected, provenanceResult);
+        }
+
+        private static void AssertProvenanceResult(List<Tuple<string, Operation, Provenance, int>> expected, string project, string itemValue, int position)
+        {
+            var p = ObjectModelHelpers.CreateInMemoryProject(project);
+            var item = p.Items.Where(i => i.EvaluatedInclude.Equals(itemValue)).ElementAt(position);
+
+            var provenanceResult = p.GetItemProvenance(item);
+            AssertProvenanceResult(expected, provenanceResult);
+        }
+
+        private static void AssertProvenanceResult(List<Tuple<string, Operation, Provenance, int>> expected, List<ProvenanceResult> actual)
```

Comment:
Definitely not a big deal as long as there's a reason. Maybe worth a comment though.